extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/EagerBuildTimeConditionBuildItem.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.arc.deployment;
+
+import org.jboss.jandex.AnnotationTarget;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * A type of build item that is similar to {@link BuildTimeConditionBuildItem} but evaluated before
+ * processing the {@link AdditionalBeanBuildItem} in order to filter the beans thanks to build time conditions
+ * before actually adding them with a {@link AdditionalBeanBuildItem}.
+ *
+ * @see io.quarkus.arc.deployment.BuildTimeConditionBuildItem
+ * @see io.quarkus.arc.deployment.AdditionalBeanBuildItem
+ */
+public final class EagerBuildTimeConditionBuildItem extends MultiBuildItem {
```

Comment:
I don't really like the name, it is misleading I think.

Unfortunately I don't really have an alternative to offer, I just know that `Eager`  isn't what we want :)